pkg/controller/clone/planner.go

```diff
@@ -410,6 +481,131 @@ func (p *Planner) planHostAssistedFromPVC(ctx context.Context, args *PlanArgs) (
 	return []Phase{hcp, rp}, nil
 }
 
+func (p *Planner) planHostAssistedFromSnapshot(ctx context.Context, args *PlanArgs) ([]Phase, error) {
+	sourceSnapshot := &snapshotv1.VolumeSnapshot{}
+	exists, err := getResource(ctx, p.Client, args.DataSource.Namespace, args.DataSource.Spec.Source.Name, sourceSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("source claim does not exist")
+	}
+
+	vsc, err := GetCompatibleVolumeSnapshotClass(ctx, p.Client, args.TargetClaim)
+	if err != nil {
+		return nil, err
+	}
+	if vsc == nil {
+		return nil, fmt.Errorf("no compatible volumesnapshotclass")
+	}
+
+	// TODO: Maybe improve the logic to create the sourceClaim for host assisted clone, or even just make it more pretty
+	sourceClaimForDumbClone := createDesiredClaim(args.DataSource.Namespace, args.TargetClaim)
```

Comment:
Pretty sure you are going to have to set a different storage class on this thing...one that is compatible with the source.  Otherwise we wouldn't be executing this strategy